### PR TITLE
Test container example for ImmuDB

### PR DESCRIPTION
**Pull request by hariohmprasath**
## Description of the change:

Added an example for running tests using `ImmuDB` database using `testcontainers`. Tests performs all the basic operations performed on the `ImmuDB` #5594 

We already have a ask for this example, so contributed the same using this pull request:
https://github.com/testcontainers/testcontainers-java/issues/5594

### CODE REVIEW

File: examples/immudb/src/test/java/ImmuDbTest.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    @Test
+    public void testGetValue() {
+        try {
+            this.immuDbClient.putValue("test1", "test2");
+            Assert.assertEquals("test2", this.immuDbClient.getValue("test1"));
+        }
+        catch (CorruptedDataException | VerificationException e) {
+            Assert.fail();
+        }
+    }
```
Comment (eddumelendez) [inline examples/immudb/src/test/java/ImmuDbTest.java:80]: we can keep this test. Given the purpose is to demonstrate how to start the container and build the client, this test will be more than enough. Also, use `assertThat` from `Assertions` (assertj)

File: examples/immudb/build.gradle [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    compileOnly 'org.slf4j:slf4j-api:2.0.0'
+    implementation 'io.codenotary:immudb4j:0.9.10.2'
+    testImplementation 'org.testcontainers:testcontainers'
+    testImplementation 'org.testng:testng:7.5'
```
Comment (eddumelendez) [inline examples/immudb/build.gradle:13]: remove this dependency

**Review by eddumelendez — CHANGES_REQUESTED** [review]
thanks for the PR @hariohmprasath ! Few notes, `ImmuDbClient` and `ImmuDbClientImpl` can be removed and the client can be used directly in the test class. Also, add the example [here](https://github.com/testcontainers/testcontainers-java/blob/master/docs/examples.md)

**Comment by hariohmprasath** [comment]
> thanks for the PR @hariohmprasath ! Few notes, `ImmuDbClient` and `ImmuDbClientImpl` can be removed and the client can be used directly in the test class. Also, add the example [here](https://github.com/testcontainers/testcontainers-java/blob/master/docs/examples.md)

Thanks @eddumelendez for the review. I have addressed all the feedback, let me know if you any suggestions. Thanks

**Comment by eddumelendez** [comment]
Add the example [here](https://github.com/testcontainers/testcontainers-java/blob/master/docs/examples.md), please.

**Comment by hariohmprasath** [comment]
> Add the example [here](https://github.com/testcontainers/testcontainers-java/blob/master/docs/examples.md), please.

Sorry about missing that in my previous commit. I have updated now, so we should good. Thanks

File: examples/immudb/src/test/java/ImmuDbTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    // Default username for the ImmuDb server
+    private final String IMMUDB_USER = "immudb";
+    // Default password for the ImmuDb server
+    private final String IMMUDB_PASSWORD = "immudb";
+    // Default database name for the ImmuDb server
+    private final String IMMUDB_DATABASE = "defaultdb";
+
+    // Test container for the ImmuDb database, with the latest version of the image and exposed port
+    @ClassRule
+    public static final GenericContainer<?> immuDbContainer = new GenericContainer<>("codenotary/immudb:latest")
```
Comment (eddumelendez) [inline examples/immudb/src/test/java/ImmuDbTest.java:29]: last one... do not use latest :)

Comment (hariohmprasath) [reply examples/immudb/src/test/java/ImmuDbTest.java]: No problem, updated the image tag to 1.3

File: examples/immudb/src/test/java/ImmuDbTest.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    public static final GenericContainer<?> immuDbContainer = new GenericContainer<>("codenotary/immudb:1.3")
+        .withExposedPorts(IMMUDB_PORT).waitingFor(
+            Wait.forLogMessage(".*Web API server enabled.*", 1)
+        );
+    // ImmuClient used to interact with the DB
+    private ImmuClient immuClient;
+
+    /**
+     * Set up the test, by creating the ImmuDbClient and connecting to the DB.
+     */
```
Comment (eddumelendez) [inline examples/immudb/src/test/java/ImmuDbTest.java:38]:
```suggestion
```

**Review by eddumelendez — APPROVED** [review]

**Comment by eddumelendez** [comment]
Thanks for your contribution, @hariohmprasath ! This is not in `main` branch

**Pull request merged**